### PR TITLE
Cmm sendqueue

### PR DESCRIPTION
**Pull request by chmorgan**

### CODE REVIEW

**Comment by chmorgan** [comment]
@kayoub5 any ideas why the TestTransmitEmpty test is failing? Its returning an odd value, some really large, but not negative, value. Passes here but not on npcap. I'm not sure how to debug but I'm able to reproduce here on Windows. 

I was also wondering if we should check the values returned from the other transmit calls. I know we are getting the correct number of packets as the tests are passing but it would be another helpful check.

**Comment by chmorgan** [comment]
@kayoub5 and I've been going through the libpcap and npcap code and haven't spotted any case where it makes sense for this crazy return value. It looks like an uninitialized value actually.

File: SharpPcap/LibPcap/SendQueue.cs [hunk 1]
```diff
@@ -209,7 +219,9 @@ protected unsafe int NativeTransmit(PcapDevice device, SendQueueTransmitModes tr
                     len = (uint)CurrentLength,
                     ptrBuff = new IntPtr(buf)
                 };
-                return LibPcapSafeNativeMethods.pcap_sendqueue_transmit(device.PcapHandle, ref pcap_queue, sync);
+                var retval = LibPcapSafeNativeMethods.pcap_sendqueue_transmit(device.PcapHandle, ref pcap_queue, sync);
+                Console.WriteLine("retval {3}, CurrentLength {0}, buffer.Length {1}, ptrBuff {2}", CurrentLength, buffer.Length, pcap_queue.ptrBuff, retval);
+                return retval;
```
Comment (kayoub5) [inline SharpPcap/LibPcap/SendQueue.cs:224]: Don't forget to revert this once you are done debugging

File: SharpPcap/LibPcap/SendQueue.cs [hunk 2]
```diff
@@ -200,6 +204,12 @@ protected unsafe int ManagedTransmit(PcapDevice device, SendQueueTransmitModes t
 
         protected unsafe int NativeTransmit(PcapDevice device, SendQueueTransmitModes transmitMode)
         {
+            if (CurrentLength == 0)
+            {
+                // Npcap does not properly check for 0 packets queue
+                // See https://github.com/nmap/npcap/issues/287
+                return 0;
```
Comment (kayoub5) [inline SharpPcap/LibPcap/SendQueue.cs:211]: @chmorgan this should fix the test

Comment (chmorgan) [reply SharpPcap/LibPcap/SendQueue.cs:211]: thanks man

**Review by kayoub5 — APPROVED** [review]

**Pull request merged**